src/openforms/submissions/form_logic.py

```diff
@@ -186,21 +188,22 @@ def evaluate_form_logic(
         # debounced client-side data changes
         data_diff = {}
 
-        for key, new_value in step.data.items():
-            original_value = initial_data.get(key)
+        for component in iter_components(configuration):
```

Comment:
I requested an inline comment to be added why we iterate over all the components rather than over `step.data`.

Reply:
@dariomory 

```suggestion
        # Iterate over all components instead of `step.data`, to take hidden fields into account (See: #1755)
        for component in iter_components(configuration):
```